src/ServiceManagement/Compute/Commands.ServiceManagement/Model/RoleConfiguration.cs

```diff
@@ -104,8 +105,18 @@ internal XElement Serialize()
             {
                 XElement certificateElement = new XElement(this.ns + "Certificate");
                 certificateElement.SetAttributeValue("name", certificate.Key);
-                certificateElement.SetAttributeValue("thumbprint", certificate.Value.Thumbprint);
-                certificateElement.SetAttributeValue("thumbprintAlgorithm", certificate.Value.ThumbprintAlgorithm);
+                if (certificate.Value.Thumbprint != null)
```

Comment:
Could certificate.Value be null?

Reply:
No. It's a value driven by the foreach loop over existing elements.

Reply:
I meant this.Certificates is of type Dictionary<string, CertificateConfiguration>();. and the for loop make sure that variable "certificate" is not null... I wonder if "certificate.Value' can be null.

Reply:
Oh, I see what you're asking.
Certificates.value won't be null because in the deserializer where it's set up, an element to the Dictionary is only added after the CertificateConfiguration object is created using an Object Initializer (so it can't be null).
See lines 44-53.